src/dist/replication/lib/replica_split.cpp

```diff
@@ -606,6 +606,183 @@ void replica::parent_check_sync_point_commit(decree sync_point) // on primary pa
+                    "current status should be inactive, wrong status = {}",
+                    enum_to_string(status()));
+
+    ddebug_replica(
+        "send register child({}) request to meta_server, current ballot = {}, child ballot = {}",
+        request.child_config.pid,
+        request.parent_config.ballot,
+        request.child_config.ballot);
+
+    rpc_address meta_address(_stub->_failure_detector->get_servers());
+    std::unique_ptr<register_child_request> req = make_unique<register_child_request>(request);
+    register_child_rpc rpc(std::move(req), RPC_CM_REGISTER_CHILD_REPLICA);
+    _primary_states.register_child_task =
+        rpc.call(meta_address,
+                 tracker(),
+                 [this, rpc](error_code ec) mutable {
+                     on_register_child_on_meta_reply(ec, rpc.request(), rpc.response());
+                 },
+                 _split_states.parent_gpid.thread_hash());
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica::on_register_child_on_meta_reply(
+    dsn::error_code ec,
+    const register_child_request &request,
+    const register_child_response &response) // on primary parent
+{
+    FAIL_POINT_INJECT_F("replica_on_register_child_on_meta_reply", [](dsn::string_view) {});
+
+    _checker.only_one_thread_access();
+
+    // primary parent is under reconfiguration, whose status should be PS_INACTIVE
+    if (partition_status::PS_INACTIVE != status() || _stub->is_connected() == false) {
+        dwarn_replica("status wrong or stub is not connected, status = {}",
+                      enum_to_string(status()));
+        _primary_states.register_child_task = nullptr;
+        // TODO(heyuchen): TBD - clear other split tasks in primary context
+        return;
+    }
+
+    if (ec != ERR_OK || (ec == ERR_OK && response.err != ERR_OK)) {
+        dsn::error_code err = ec == ERR_OK ? response.err : ec;
+        dwarn_replica(
+            "register child({}) failed, error = {}, request child ballot = {}, local ballot = {}",
+            request.child_config.pid,
+            err.to_string(),
+            request.child_config.ballot,
+            get_ballot());
+
+        // register request is out-of-dated
+        if (err == ERR_INVALID_VERSION) {
+            return;
+        }
+
+        // we need not resend register request if child has been registered
+        if (err != ERR_CHILD_REGISTERED) {
+            _primary_states.register_child_task =
+                tasking::enqueue(LPC_DELAY_UPDATE_CONFIG,
+                                 tracker(),
+                                 std::bind(&replica::parent_send_register_request, this, request),
+                                 get_gpid().thread_hash(),
+                                 std::chrono::seconds(1));
+            return;
+        }
+    }
+
+    // meta_server error
+    if (response.parent_config.pid != get_gpid() || response.child_config.pid != _child_gpid) {
```

Comment:
I think you can move this code block before ` if (ec != ERR_OK || (ec == ERR_OK && response.err != ERR_OK)) {`.
So these code can be changed like this:
```
   // meta_server error
    if (response.parent_config.pid != get_gpid() || response.child_config.pid != _child_gpid) {
          ......
    }

   if (err != ERR_OK) {
       .......
   } else {
       ......
   }
```